### DISCUSSION TITLE
MassDistribution vs MergerTreeBuildMasses

### DISCUSSION THREAD

**Discussion by Sam230685** [galacticusorg/galacticus#643] [Q&A]
Hello again. What is the difference between 3.84 Merger Tree Build Masses (specifically default sampledDistributionUniform) and 3.76 Mass Distributions?

**Comment by abensonca** [comment]
The `mergerTreeBuildMasses` class is what actually specifies the masses of merger trees to be built. For example, using in your parameter file:
```
<mergerTreeBuildMasses value="fixedMass">
.
.
.
</mergerTreeBuildMasses>
```
allows specifying a fixed mass of tree.

Some members of the `mergerTreeBuildMasses` class, e.g. `sampledDistributionUniform` (which is the default), will choose masses of trees by drawing them from a distribution. In that case, the distribution from which to draw masses is specified by the `mergerTreeBuildMassDistribution` class (section 3.95).

The `massDistribution` class (section 3.76) is a completely different thing. It's used to define the spatial distribution of mass in various components of a galaxy/halo. For example, `massDistributionNFW` defines the NFW dark matter halo density profile (and computes things such as the mass enclosed within a given radius in that profile, the gravitational potential, etc.).

**Reply by Sam230685** [reply]
So for example if I wanted to implement a massDistribution that is constantDensityCloud I would use:
```
 <massDistributionDisk value="constantDensityCloud">
	<mass value="some amount of mass"/>
 </massDistributionDisk>
```
with the amount of mass being within however much is specified by the merger tree file?

**Comment by abensonca** [comment]
This wouldn't work as the disk is required to be spherically symmetric. And, the mass of any disk isn't specified in the parameter file in this way, rather it is computed, or specified as part of an initial conditions file.

It sounds like you're trying to do something relatively non-standard with Galacticus right away. It can probably do what you want to do, but you'll need a deeper understanding of how it all works to make it do it. If you can give a more detailed explanation of _exactly_ what you want to achieve I can probably help you set up a minimal example and explain how it works.

**Reply by Sam230685** [reply]
I would like to see how a distribution of stars is impacted by the initial conditions of the gas cloud. The main things I would like to achieve are the following.
1) Manipulate mass density/amount of initial cloud within a fairly simple geometry. Along with testing for different distributions, not just uniform. 
2) Start with only gas cloud, which I have already seen closedBoxTree.xml on how to do this.
3) Output data in order to track star formation (I know that you said Galacticus did not track individual star masses/types so the stellar mass of the galaxy is a fine alternative, also star formation rate would be useful)
4) And obviously I would like to be able to change initial conditions in order to build my dataset.

My main struggles are currently getting the proper spatial mass distribution along with getting the right outputs. I am currently working using the closedBoxTree.xml as my merger tree file and my current parameter file is below. Thanks for the help. 

```
<?xml version="1.0" encoding="UTF-8"?>
<!-- Test of closed-box chemical evolution -->
<!-- 08-July-2015                          -->
<parameters>
  <formatVersion>2</formatVersion>
  <version>0.9.4</version>

  <!-- Component selection -->
  <componentBasic value="standard"/>
  <componentHotHalo value="null"/>
  <componentBlackHole value="null"/>
  <componentDarkMatterProfile value="null"/>
  <componentDisk value="standard"/>
  
  <componentSatellite value="null"/>
  <componentSpheroid value="null">
  </componentSpheroid>
  <componentSpin value="scalar"/>
  <!-- Cosmological parameters and options -->
  <cosmologyFunctions value="matterLambda"/>
  <cosmologyParameters value="simple">
    <HubbleConstant value="70.0"/>
    <OmegaMatter value="0.3"/>
    <OmegaDarkEnergy value="0.7"/>
    <OmegaBaryon value="0.05"/>
    <temperatureCMB value="2.72548"/>
  </cosmologyParameters>

  <!-- Power spectrum options -->
  <cosmologicalMassVariance value="peakBackgroundSplit">
  </cosmologicalMassVariance>
  <transferFunction value="eisensteinHu1999">
    <neutrinoNumberEffective value="3.046"/>
    <neutrinoMassSummed value="0.000"/>
  </transferFunction>
  <powerSpectrumPrimordial value="powerLaw">
    <index value="1.000"/>
    <wavenumberReference value="1.000"/>
    <running value="0.000"/>
  </powerSpectrumPrimordial>

  <!-- Structure formation options -->
  <linearGrowth value="collisionlessMatter"/>
  <haloMassFunction value="Tinker2008"/>
  <criticalOverdensity value="sphericalCollapseClsnlssMttrCsmlgclCnstnt"/>
  <virialDensityContrast value="sphericalCollapseClsnlssMttrCsmlgclCnstnt"/>
  
 
  <powerSpectrumNonlinear value="smith2003"/>
  <!-- Merger tree building options -->
  <mergerTreeConstructor value="fullySpecified">
    <fileName value="closedBoxTree.xml"/>
  </mergerTreeConstructor>
  <!-- Halo accretion options -->
  <accretionHalo value="zero"/>

  <!-- Dark matter halo profile -->
  <darkMatterProfileDMO value="isothermal"/>

  <!-- Hot halo model options -->
  <hotHaloMassDistribution value="null"/>

  <!-- Galactic structure options -->
  <galacticStructureSolver value="fixed"/>

  <!-- Star formation rate -->
  <starFormationRateDisks value="timescale">
    <starFormationTimescale value="fixed">
      <timescale value="0.25"/>
    </starFormationTimescale>
  </starFormationRateDisks>

  <!-- Metal production -->
  <stellarPopulation value="standard">
    <metalYield value="0.025"/>
    <recycledFraction value="0.400"/>
  </stellarPopulation>

  <!-- Node evolution and physics -->
  <nodeOperator value="multi">
    <!-- Cosmological epoch -->
    <nodeOperator value="cosmicTime"/>
    <!-- DMO evolution -->
    <nodeOperator value="DMOInterpolate"/>
    <!-- Halo spins -->
    <nodeOperator value="haloAngularMomentumInterpolate"/>
    <!-- Satellite evolution -->
    <nodeOperator value="satelliteMergingTime"/>
    <nodeOperator value="satelliteMassLoss"/>
    <!-- Star formation -->
    <nodeOperator value="starFormationDisks"/>
  </nodeOperator>
  
  <!-- Numerical tolerances -->
  <mergerTreeNodeEvolver value="standard">
    <odeToleranceAbsolute value="1.0e-6"/>
    <odeToleranceRelative value="0.001"/>
  </mergerTreeNodeEvolver>

  <mergerTreeEvolver value="standard">
    <timestepHostAbsolute value="1.000"/>
    <timestepHostRelative value="0.100"/>
  </mergerTreeEvolver>

  <!-- Output options -->
  <mergerTreeOutputter value="multi">
    <mergerTreeOutputter value="standard">
		<outputReferences value="false"/>
	</mergerTreeOutputter>
	<mergerTreeOutputter value="analyzer"/>
  </mergerTreeOutputter>
  <!--
  <outputAnalysis value="massFunctionStellarSDSS"/>
  <outputTimes value="list">
	<redshifts value="0.0 0.06"/>
  </outputTimes>
	-->
	
  <!-- Output file -->
  <outputFileName value="output.hdf5"/>

</parameters>

```

**Reply by abensonca** [reply]
I see. In that case I don't think Galacticus is the tool you are looking for. A semi-analytic model like Galacticus typically won't compute the star formation rate dependent on arbitrary mass distributions. Instead, they typically employ simple models relating star formation rate to global properties of a galaxy.

You may want to consider looking into running hydrodynamical simulations with star formation physics included to achieve your goals.
